### PR TITLE
Another round of small fixes

### PR DESCRIPTION
**Pull request by kilemensi**
## Description

 - [x] Allow content editors to change logo via CMS,
 - [x] Load NewsletterSubscription dynamically without SSR because the script assumes windows is present,
 - [x] Fix missing id on grantees block, and
 - [x] Ensure canonical url respects current local.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Screenshots

Screenshot 2023-07-14 at 17 01 56

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by kilemensi** [comment]
@koechkevin / @kelvinkipruto

File: apps/charterafrica/src/lib/data/common/index.js [hunk 1]
```diff
@@ -17,25 +17,38 @@ import processPagePrivacyPolicy from "@/charterafrica/lib/data/common/processPag
...
 import { getPageSeoFromMeta } from "@/charterafrica/lib/data/seo";
 
+function getLogoProps(logo) {
+  const {
+    href = "/",
+    image: {
+      alt = "Charter Africa",
+      fill = true,
+      priority = true,
+      src = "/images/charter-logo.svg",
```
Comment (kelvinkipruto) [inline apps/charterafrica/src/lib/data/common/index.js:27]:
```suggestion
      url:src = "/images/charter-logo.svg",
```

File: apps/charterafrica/src/components/Documents/Documents.js [hunk 2]
```diff
@@ -85,6 +87,7 @@ const Documents = React.forwardRef(function Documents(props, ref) {
       setDocuments(foundDocuments);
       setPinnedDocuments(newPinnedDocuments);
       setPage(currentPage);
+      setDocuments(foundDocuments);
```
Comment (kelvinkipruto) [inline apps/charterafrica/src/components/Documents/Documents.js:90]: This is already called here [here](https://github.com/CodeForAfrica/ui/blob/2c8a45bad3b859b81e7da1ea32c56788b3929061/apps/charterafrica/src/components/Documents/Documents.js#L87)

File: apps/charterafrica/src/payload/globals/Navigation.js [hunk 3]
```diff
@@ -38,6 +43,40 @@ const Navigation = {
...
+          filterOptions: {
+            mimeType: { contains: "image" },
+          },
+          label: {
+            en: "Image",
+            pt: "Imagem",
+            fr: "Logo",
+          },
+        },
+        link({
```
Comment (kelvinkipruto) [inline apps/charterafrica/src/payload/globals/Navigation.js:69]: Not sure why we need to link this to any other page other than `/`

File: apps/charterafrica/src/lib/data/common/index.js (hunk 1 above)
Comment (kilemensi) [reply apps/charterafrica/src/lib/data/common/index.js:27]: What is url:src?

File: apps/charterafrica/src/payload/globals/Navigation.js (hunk 3 above)
Comment (kilemensi) [reply apps/charterafrica/src/payload/globals/Navigation.js:69]: 👇🏽 is how the NavBar looks in prod

Screenshot 2023-07-17 at 12 55 37

Doesn't have to but `logo` and `HOME` could point to different links.

File: apps/charterafrica/src/components/Documents/Documents.js (hunk 2 above)
Comment (kilemensi) [inline apps/charterafrica/src/components/Documents/Documents.js:90]:
```suggestion
```

File: apps/charterafrica/src/components/DesktopNavBar/DesktopNavBar.js [hunk 4]
```diff
@@ -18,7 +18,7 @@ const DesktopNavBar = React.forwardRef(function DesktopNavBar(props, ref) {
       sx={sx}
     >
       <Grid item>
-        <Logo {...logo} width={230} height={58} />
+        <Logo {...logo} sx={{ height: 59, width: 230 }} />
```
[the remark this answers is not in the document]
Comment (kilemensi) [reply apps/charterafrica/src/components/DesktopNavBar/DesktopNavBar.js:21]: These come from Figma so we can discuss it with the designer

File: apps/charterafrica/src/lib/data/common/index.js (hunk 1 above)
Comment (kelvinkipruto) [reply apps/charterafrica/src/lib/data/common/index.js:27]: Changing the logo in cms does not update the image logo on the front page as `src=/images/charter-logo.svg`. The payload from the CMS has the image source as `url`.
So we need to rename the `url` to `src` and then default it to `/images/charter-logo.svg`
Here is a sample payload: 
```
image: {
      id: '641a034566df56a12ded82ae',
      alt: 'People',
      prefix: 'media',
      filename: 'people-icon.svg',
      mimeType: 'image/svg+xml',
      filesize: 1008,
      width: 125,
      height: 124,
      createdAt: '2023-03-21T19:19:33.269Z',
      updatedAt: '2023-03-21T19:19:33.269Z',
      url: 'http://localhost:3000/media/people-icon.svg'
    },
   linkType: 'internal',
    doc: { value: [Object], relationTo: 'pages' },
    href: '/opportunities/helpdesk'
 }
```

Comment (kilemensi) [reply apps/charterafrica/src/lib/data/common/index.js:27]: Oh... good catch but you can't change that via `url:src? = `.

**Review by koechkevin — APPROVED** [review]

**Pull request merged**